### PR TITLE
python310Packages.rsa: update and fix tests

### PR DESCRIPTION
**Pull request by tjni**
###### Description of changes

We can get rid of old Python 2 dependencies and enable the tests by fetching from GitHub.

Because this gets rid of unittest2, it's related to https://github.com/NixOS/nixpkgs/pull/154264 and https://github.com/NixOS/nixpkgs/pull/154650.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/rsa/default.nix [hunk 1]
```diff
@@ -1,36 +1,47 @@
...
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
   propagatedBuildInputs = [ pyasn1 ];
 
-  preConfigure = lib.optionalString (isPy3k && pythonOlder "3.7") ''
+  checkInputs = [
+    mypy
+    pytest-cov
```
Comment (fabaff) [inline pkgs/development/python-modules/rsa/default.nix:34]: PLease patch out the coverage parts. This doesn't add value for us.

Comment (tjni) [reply pkgs/development/python-modules/rsa/default.nix]: Done, thanks for the feedback.

File: pkgs/development/python-modules/rsa/default.nix [hunk 2]
```diff
@@ -1,36 +1,49 @@
...
-  preConfigure = lib.optionalString (isPy3k && pythonOlder "3.7") ''
-    substituteInPlace setup.py --replace "open('README.md')" "open('README.md',encoding='utf-8')"
+  preCheck = ''
+    sed -i '/addopts/d' tox.ini
   '';
 
-  # No tests in archive
-  doCheck = false;
+  checkInputs = [
+    mypy
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/rsa/default.nix:36]: Can we easily skip typechecks?

Comment (tjni) [reply pkgs/development/python-modules/rsa/default.nix]: Yes, it's possible. I disabled `tests/test_mypy.py`.

**Pull request merged**